File: aihub-python/src/aihub/models/model_config.py
from sqlalchemy import Column, Integer, String, Text, DateTime
from sqlalchemy.sql import func
from .database import Base


class ModelConfig(Base):
    __tablename__ = "model_config"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String(255), nullable=False)
    vendor = Column(String(50), nullable=False)
    model_id = Column(String(100), nullable=False)
    api_key = Column(String(500), nullable=False)
    base_url = Column(String(255))
    status = Column(Integer, default=1)
    config = Column(Text)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    is_deleted = Column(Integer, default=0)

File: aihub-java/aihub-admin/src/main/java/com/aihub/admin/entity/ModelConfig.java
package com.aihub.admin.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 模型配置实体类
 */
@Data
@TableName("model_config")
public class ModelConfig {

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 模型名称
     */
    private String name;

    /**
     * 厂商 OpenAI/Claude/DeepSeek等
     */
    private String vendor;

    /**
     * 模型ID
     */
    private String modelId;

    /**
     * API Key（加密存储）
     */
    private String apiKey;

    /**
     * Base URL
     */
    private String baseUrl;

    /**
     * 状态 0-禁用 1-启用
     */
    private Integer status;

    /**
     * 模型配置参数（JSON格式）
     */
    private String config;

    @TableField("created_at")
    private LocalDateTime createdAt;

    @TableField("updated_at")
    private LocalDateTime updatedAt;

    @TableField("is_deleted")
    private Integer isDeleted;

    /**
     * 是否为默认模型
     */
    @TableField("is_default")
    private Integer isDefault;

    /**
     * 模型类型: chat=对话模型, embedding=向量模型, image=文生图模型, audio=语音模型, rerank=重排序模型
     */
    @TableField("model_type")
    private String modelType;
}
